tests/unit/db/test_ingest.py

```diff
@@ -26,55 +37,198 @@ def test_cve_data_ingested(self, cve_examples: Path) -> None:
         for cve in nvddb.cve_all():
             assert isinstance(cve, CVE)
 
-    @pytest.mark.parametrize(
-        ("offset", "last_mod_start_date"),
-        [
-            (None, None),
-            (0, None),
-            (42, None),
-            (None, datetime.now(timezone.utc)),
-            (42, datetime.now(timezone.utc)),
-        ],
-    )
-    def test_cve_api_args(self, offset: int | None, last_mod_start_date: datetime | None) -> None:
+    @pytest.mark.parametrize("update", [False, False, False, True, True])
+    def test_cve_api_args(self, update: bool) -> None:
         with patch("bomsquad.vulndb.db.ingest.NVD.vulnerabilities") as vulns:
-            if offset:
-                Ingest.cve(offset, last_mod_start_date=last_mod_start_date)
-            else:
-                Ingest.cve(last_mod_start_date=last_mod_start_date)
+            Ingest.cve(update=update)
             assert vulns.call_count == 1
             args, kwargs = vulns.call_args
-            if offset:
-                assert args[0] == offset
+            assert kwargs["offset"] == 0
+            cp = Checkpoints()
+            if update:
+                assert kwargs["last_mod_start_date"] == cp.last_updated("cve")
             else:
-                assert args[0] == 0
-            assert kwargs["last_mod_start_date"] == last_mod_start_date
+                assert kwargs["last_mod_start_date"] is None
 
     def test_cpe_data_ingested(self, cpe_examples: Path) -> None:
         assert nvddb.cpe_count() == len(list(cpe_examples.iterdir()))
         for cpe in nvddb.cpe_all():
             assert isinstance(cpe, CPE)
 
-    @pytest.mark.parametrize(
-        ("offset", "last_mod_start_date"),
-        [
-            (None, None),
-            (0, None),
-            (42, None),
-            (None, datetime.now(timezone.utc)),
-            (42, datetime.now(timezone.utc)),
-        ],
-    )
-    def test_cpe_api_args(self, offset: int | None, last_mod_start_date: datetime | None) -> None:
+    @pytest.mark.parametrize("update", [False, False, False, True, True])
+    def test_cpe_api_args(self, update: bool) -> None:
         with patch("bomsquad.vulndb.db.ingest.NVD.products") as products:
-            if offset:
-                Ingest.cpe(offset, last_mod_start_date=last_mod_start_date)
-            else:
-                Ingest.cpe(last_mod_start_date=last_mod_start_date)
+            Ingest.cpe(update=update)
             assert products.call_count == 1
             args, kwargs = products.call_args
-            if offset:
-                assert args[0] == offset
+            assert kwargs["offset"] == 0
+            cp = Checkpoints()
+            if update:
+                assert kwargs["last_mod_start_date"] == cp.last_updated("cpe")
             else:
-                assert args[0] == 0
-            assert kwargs["last_mod_start_date"] == last_mod_start_date
+                assert kwargs["last_mod_start_date"] is None
+
+
```

Comment:
These test cases test the functionality of both db/ingest.py and client/nvd.py, but their purpose is to target cve and cpe methods in src/bomsquad/vulndb/db/ingest.py . Instead of moving the whole code to client/test_nvd.py , I will just split it into two unit tests: One testing only the functionality of NVD.vulnerabilities and NVD.products (patching requests), and the other one only testing the functionality of Ingest API (patching NVD.vulnerabilities and NVD.products).

Reply:
Yes - that's basically what I mean ... I don't think there need to be substantial changes to the ingest tests if you hide most of this behind the NVD APIs - I called out the need for some 'update during ingest' simulation in the ingest tests in another comment.